src/Microsoft.AspNetCore.Mvc.ViewFeatures/ViewFeatures/HtmlHelper.cs

```diff
@@ -589,7 +589,15 @@ public Task RenderPartialAsync(string partialViewName, object model, ViewDataDic
                 // Determine which ViewData we should use to construct a new ViewData
                 var baseViewData = viewData ?? ViewData;
 
-                var newViewData = new ViewDataDictionary(baseViewData, model);
+                // Make an identical, defensive copy of the ViewDataDictionary to isolate changes.
+                ViewDataDictionary newViewData = new ViewDataDictionary(baseViewData);
```

Comment:
Opps, I temporarily had an `if` and `else` that assigned to this variable. Forgot to revert the type.